### PR TITLE
[RDY] 1326 fix using dofile instead of requrire

### PR DESCRIPTION
**Pull request by pschoffer**
This is related to #1326. Please see the ticket for more info.

Also we have removed require from the strict mode, which is only really needed for the Cpp modules. In my change I dropped that change and I am using non strict version of `require` only when needed.

### CODE REVIEW

File: CorsixTH/CorsixTH.lua [hunk 1]
```diff
@@ -56,13 +56,16 @@ function dofile(name)
   return save_results(done_files, name, persist.dofile(code_dir .. name .. ".lua"))
 end
 
+-- Adds code dir as a first place to look for required files
+package.path = code_dir .. "?.lua" .. ";" .. package.path
```
Comment (TheCycoONE) [inline CorsixTH/CorsixTH.lua]: also add code_dir .. "?" .. pathsep .. "init.lua" for completeness - lua normally allows that naming convention.

**Comment by TheCycoONE** [comment]
This looks close to ready, and I would like to get it merged sooner than later. Would you mind squishing it into 2 commits? (first 2 functional commits into one, second 2 style commits into another) ... if you do my path change to include ?/init.lua put that in the first commit.

If you don't feel comfortable squishing but are otherwise ready to merge let me know and I'll handle it.

**Comment by pschoffer** [comment]
@TheCycoONE I squashed the commit, but I didn't know what you meant by `code_dir .. "?" .. pathsep .. "init.lua"`. As far as I know lua replaces `?` with what you put to require. So if you would put `require("cool_thing")` it would look for `code_dir/cool_thing/init.lua` is this what you meant?

**Comment by TheCycoONE** [comment]
Exactly.

**Comment by pschoffer** [comment]
I tested it with `require("blbe")` and got 
```
An error has occurred in CorsixTH:
./CorsixTH/CorsixTH.lua:36: module 'blbe' not found:
	no field package.preload['blbe']
	no file './CorsixTH/Lua/blbe.lua'
	no file './CorsixTH/Lua/blbe/init.lua'
	no file '/usr/local/share/lua/5.2/blbe.lua'
	no file '/usr/local/share/lua/5.2/blbe/init.lua'
	no file '/usr/local/lib/lua/5.2/blbe.lua'
	no file '/usr/local/lib/lua/5.2/blbe/init.lua'
	no file '/usr/share/lua/5.2/blbe.lua'
	no file '/usr/share/lua/5.2/blbe/init.lua'
	no file './blbe.lua'
	no file '/usr/local/lib/lua/5.2/blbe.so'
	no file '/usr/lib/x86_64-linux-gnu/lua/5.2/blbe.so'
	no file '/usr/lib/lua/5.2/blbe.so'
	no file '/usr/local/lib/lua/5.2/loadall.so'
	no file './blbe.so'
stack traceback:
	[C]: in function 'require'
	./CorsixTH/CorsixTH.lua:36: in main chunk
	[C]: in ?
```

So seems to work as expected

**Comment by TheCycoONE** [comment]
+1. Assuming you're ready, mark as [RDY] please :)

**Pull request merged**